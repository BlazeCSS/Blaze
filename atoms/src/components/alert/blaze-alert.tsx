import { Component, Method, Prop, State } from '@stencil/core';

@Component({
  tag: 'blaze-alert'
})
export class Alert {

  @Prop() type: string;
  @Prop() dismissible: boolean = false;
  @Prop() open: boolean = false;
  @State() _isOpen: boolean = false;

  @Method()
  close() {
    this._isOpen = false;
  }

  @Method()
  show() {
    this._isOpen = true;
  }

  @Method()
  isOpen() {
    return this._isOpen;
  }

  componentWillLoad() {
    this._isOpen = this.open;
  }

  render() {
    const isOpenClass = !this._isOpen ? 'u-display-none' : '';

    return (
      <div class={`c-text c-alert c-alert--${this.type} ${isOpenClass}`}>
        {
          this.dismissible &&
          <button class="c-button c-button--close" onClick={() => this.close()}>
            &times;
          </button>
        }
        <slot />
      </div>
    );
  }
}
