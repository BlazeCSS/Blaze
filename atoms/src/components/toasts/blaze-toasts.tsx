import { Component, Prop } from '@stencil/core';

@Component({
  tag: 'blaze-toasts'
})
export class Toasts {

  @Prop() position: string = 'topright';

  render() {
    return (
      <div class={`c-alerts c-alerts--${this.position}`}>
        <slot/>
      </div>
    );
  }
}
