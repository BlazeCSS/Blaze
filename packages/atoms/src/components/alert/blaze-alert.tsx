import { Component, Method, Prop, State } from '@stencil/core';

@Component({
  tag: 'blaze-alert'
})
export class Alert {

  @Prop() type: string = '';
  @Prop() dismissible: boolean = false;
  @Prop() open: boolean = false;
  @State() _isOpen: boolean = false;

  @Method()
  close() {
    this._isOpen = false;
  }

  @Method()
  show() {
    this._isOpen = true;
  }

  @Method()
  isOpen() {
    return this._isOpen;
  }

  componentWillLoad() {
    this._isOpen = this.open;
  }

  render() {
    const isOpenClass = !this._isOpen ? 'u-display-none' : '';
    const typeClass = this.type ? `c-alert--${this.type}` : '';

    return (
      <div role="alert" hidden={!this._isOpen} class={`c-alert ${typeClass} ${isOpenClass}`}>
        {
          this.dismissible &&
          <button class="c-button c-button--close" onClick={() => this.close()}>
            &times;
          </button>
        }
        <slot />
      </div>
    );
  }
}
