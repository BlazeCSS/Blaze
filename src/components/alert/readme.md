# blaze-alerts

<!-- Auto Generated Below -->

## Properties

| Property   | Attribute  | Description | Type     | Default     |
| ---------- | ---------- | ----------- | -------- | ----------- |
| `position` | `position` |             | `string` | `undefined` |

---

_Built with [StencilJS](https://stenciljs.com/)_
