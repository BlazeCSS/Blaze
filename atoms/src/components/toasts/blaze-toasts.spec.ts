import { flush, render } from '@stencil/core/testing';
import { Toasts } from './blaze-toasts';
import { Alert } from '../alert/blaze-alert';

describe('Toasts', () => {
  it('should build', () => {
    expect(new Toasts()).toBeTruthy();
  });

  let element;

  const snapIt = (name, html) => {
    it(name, async () => {
      element = await render({
        components: [Toasts, Alert],
        html
      });
      await flush(element);

      expect(element).toMatchSnapshot();
    });
  };

  snapIt(
    'renders correctly with no props',
    `<blaze-toasts position="bottomright">
      <blaze-alert dismissible="false" type="warning">You can not remove this toast</blaze-alert>
      <blaze-alert type="info">This is some information</blaze-alert>
    </blaze-toasts>`
  );
});
