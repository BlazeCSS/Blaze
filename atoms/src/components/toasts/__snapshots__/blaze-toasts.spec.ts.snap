// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toasts renders correctly with no props 1`] = `
<blaze-toasts
  class="hydrated"
  position="bottomright"
>
  <div
    class="c-alerts c-alerts--bottomright"
  >
    <!---->
    
      
    <blaze-alert
      class="hydrated"
      dismissible="false"
      type="warning"
    >
      <div
        class="c-text c-alert c-alert--warning u-display-none"
      >
        
        <!---->
        You can not remove this toast
      </div>
    </blaze-alert>
    
      
    <blaze-alert
      class="hydrated"
      type="info"
    >
      <div
        class="c-text c-alert c-alert--info u-display-none"
      >
        
        <!---->
        This is some information
      </div>
    </blaze-alert>
    
    
  </div>
</blaze-toasts>
`;
